### ISSUE TITLE
Dependency: Stem is unmaintained!

### ISSUE THREAD

**Issue by kkarhan** [torproject/nyx#68]
Hi,
as [per your FAQ](https://nyx.torproject.org/#dependencies) Nyx requires Stem 1.5.4. or later, but as per [Stem Project Website](https://stem.torproject.org/) it is mostly unmaintained.

Given that one may want to actually use Nyx, using something that hasn't been updated in over a year and likely will break as Python updates versions, this seems to be a foreseeable issue IMHO.

The [pileup of unreplied & uncommented issues in stem's GitHub](https://github.com/torproject/stem/issues) seems to be a warning sign...

---

I don't know of any good solution to this.

Maybe a newly written tool is necessary?
After all, ``tor-arm`` was the predecessor...

Personally I'd prefer a statically-linked, self-containing binary with no outside dependencies (aside from Tor obviously!) so it and [tor](https://github.com/torproject/tor) could be installed onto [any embedded operating system](https://github.com/OS-1337/OS1337) and / or [provided as reproduceably build binary via the repo](https://github.com/OS-1337/pkgs/blob/main/docs/WISHLIST.tsv) and package manager of it.

Thanks for your time.

**Comment by MikeBishop** [comment]
In fairness, nyx has not released a new version in 4 years. One might get the impression that nyx is _also_ unmaintained.

**Comment by atagar** [comment]
Yes. Nyx and Stem are both unmaintained. I'm the author of both and ceased development of both at the same time.

**Comment by kkarhan** [comment]
Thx @atagar - and whilst I don't know why [and it's honestly none of my business!] said projects are unmaintained, I'd hope we'd find some solution.

My suggestion boils down to two options:

1. Finding a new maintainer -  @torproject-git may want to do a callout on social media [(i.e. fediverse)](https://mastodon.social/@torproject) to find one.
2. Set this project into ``archive`` and deprecate it just like  [gettor](https://github.com/torproject/gettor), and it's [web frontend](https://github.com/torproject/gettor-web). 

I'm neither the maintainer of this org nor repro and thus I'd just point out these obvious ones, so feel free to do something else if you see fit.

I'd just recommend to be upfront about it being [Abandonware](https://en.wikipedia.org/wiki/Abandonware) at this point.

I've also noted this in [Stem's issues](https://github.com/torproject/stem/issues/154) as well and no matter what is being done with [nyx](https://github.com/torproject/nyx) and/or [stem](https://github.com/torproject/stem) I do sincerely hope they get either maintained or archived so people like me who find them interesting will get a clear indication that they are unmaintained and thus should refrain from using them!

**Comment by atagar** [comment]
Stem is upfront about its development status ([see its front page](https://stem.torproject.org/)). None the less, it's still the only game in town for a Python controller library that implements Tor's whole control-spec.

Finding a proper maintainer is far easier said than done. I wrote both projects over the course of a decade. Nobody ever stepped up enough to be a viable maintainer. If someone ever does then I'd suggest looking at: https://github.com/torproject/stem/issues/97

**Comment by Skelp** [comment]
> Finding a proper maintainer is far easier said than done.

Plus, we don't want to rush things, appoint a new maintainer and end up in an xz-ssh-situation again.

**Comment by drforbin** [comment]
Atagar,

Have you found a maintainer at all?
